File: statistics/php/src/Link.php
<?php

/**
 * @Author: Admin
 * @Date:   2021-04-12 22:32:07
 * @Last Modified by:   Admin
 * @Last Modified time: 2021-04-12 22:44:38
 */
require "ILink.php";
class Node{
    var $next;
    var $content;
}
class Link implements ILink{
    var $head;
    var $pos;
    function Link(){
        $this->head = new Node();
        $this->head->next = null;
        $this->head->content = null;
        $this->pos = $this->head;
    }
    function search($element){
        $currNode = $this->head;
        while($currNode->content != $element){
            $currNode = $currNode->next;
        }
        return $currNode;
    }
    function remove($node=new Node()){
        $node->content = $node->next->content;
        $node->next->next;

    }
    function append($node=new Node()){
        $this->head->next = $node;
    }
    function update($item,$element){
        $currNode = $this->search($item);
        $currNode->content = $element;
    }
}

File: statistics/Go/src/Link.go
/*
* @Author: Admin
* @Date:   2021-05-06 14:27:53
* @Last Modified by:   admin
* @Last Modified time: 2022-04-14 20:48:40
 */
package mathlib

type Node struct {
	prev    *Node
	next    *Node
	element interface{}
}

type Link struct {
	head   *Node
	pos    *Node
	length int
}

type ILink interface {
	append(item interface{})
	remove(item interface{})
	find(item interface{})
	init() Link
}

func (l *Link) init() {
	l.head = nil
	l.pos = nil
	l.length = 0
}

/**
 * [func description]
 * @param  {interface{}} item
 * @return {[type]}   [description]
 */
func (l *Link) append(item interface{}) {
	var newNode = new(Node)
	l.pos = l.head.next
	l.length++
	l.pos = newNode
}
func (l *Link) remove(item interface{}) {
	findOne := l.find(item)
	findOne.element = findOne.next.element
	findOne.next = findOne.next.next
}
func (l *Link) find(item interface{}) Node {
	var currNode = new(Node)
	currNode = l.head.next
	for currNode.element != item && currNode != nil {
		currNode = currNode.next
	}
	return *currNode
}
